specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/storagecache.json

```diff
@@ -0,0 +1,2623 @@
+              "description": "Start time of a 30 minute weekly maintenance window",
+              "type" : "string",
+              "format" : "date-time"
+            },
+            "maintenanceStatus": {
+              "description": "Maintenance status of the amlFilesystem.",
+              "$ref": "#/definitions/amlFilesystemMaintenanceStatus"
+            },
+            "hsmSettings": {
+              "$ref": "#/definitions/amlFilesystemHSMSettings",
+              "description": "Specifies HSM settings of the amlFilesystem."
+            },
+            "namespaceHydrationStatus": {
+              "description" : "Namespace Hydration Status during creation",
+              "$ref": "#/definitions/amlFilesystemHydrationStatus"
+            }
+          },
+          "type": "object"
+        },
+        "sku": {
+          "description": "SKU for the Cache.",
+          "properties": {
+            "name": {
+              "description": "SKU name for this Cache.",
+              "type": "string"
+            }
+          },
+          "type": "object"
+        }
+      },
+      "x-ms-azure-resource": true,
+      "type": "object"
+    },
+    "amlFilesystemsListResult": {
+      "description": "Result of the request to list Caches. It contains a list of Caches and a URL link to get the next set of results.",
+      "properties": {
+        "nextLink": {
+          "description": "URL to get the next set of Cache list results, if there are any.",
+          "type": "string"
+        },
+        "value": {
+          "description": "List of Caches.",
+          "items": {
+            "$ref": "#/definitions/amlFilesystem"
+          },
+          "type": "array"
+        }
+      },
+      "type": "object"
+    },
+    "amlFilesystemIdentity": {
+      "description": "amlFilesystem identity properties.",
+      "properties": {
+        "principalId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The principal id of the amlFilesystem."
+        },
+        "tenantId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The tenant id associated with the amlFilesystem."
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of identity used for the amlFilesystem",
+          "enum": [
+            "SystemAssigned",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "amlFilesystemIdentityType",
+            "modelAsString": false
+          }
+        }
+      }
+    },
+    "amlFilesystemMaintenanceStatus": {
+      "description": "Properties describing the software upgrade state of the amlFilesystem.",
+      "properties": {
+        "currentFirmwareVersion": {
+          "description": "Version string of the firmware currently installed on this amlFilesystem.",
+          "type": "string",
+          "readOnly": true
+        },
+        "firmwareUpdateStatus": {
```

Comment:
The customer has no control over this. I prefer that we not expose this.

Reply:
In the Health we have maintenance. Is that good enough.  What if maintenance fails, never finishes.  Is there enough information when your amlFilesystem is in maintenance.  

With this gone, the entire maintenance status is just currentSoftwareVersion